From 9745fcbbc84582b295e652c5986eb112b0da2e91 Mon Sep 17 00:00:00 2001
From: Thomas Betker <thomas.betker@rohde-schwarz.com>
Date: Wed, 19 Mar 2014 22:37:44 +0100
Subject: [PATCH 269/793] spi: zynq-qspi: Fix mode handling

spi_setup() already checks spi->mode against master->mode_bits, so we
don't need to check for unexpected mode bits in xqspips_setup() or
xqspips_setup_transfer().

Signed-off-by: Thomas Betker <thomas.betker@rohde-schwarz.com>
Reviewed-by: Harini Katakam <harinik@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/spi/spi-zynq-qspi.c |   10 ----------
 1 files changed, 0 insertions(+), 10 deletions(-)

diff --git a/drivers/spi/spi-zynq-qspi.c b/drivers/spi/spi-zynq-qspi.c
index 96fde42..4541562 100644
--- a/drivers/spi/spi-zynq-qspi.c
+++ b/drivers/spi/spi-zynq-qspi.c
@@ -456,12 +456,6 @@ static int zynq_qspi_setup_transfer(struct spi_device *qspi,
 
 	req_hz = (transfer) ? transfer->speed_hz : qspi->max_speed_hz;
 
-	if (qspi->mode & ~MODEBITS) {
-		dev_err(&qspi->dev, "%s, unsupported mode bits %x\n",
-			__func__, qspi->mode & ~MODEBITS);
-		return -EINVAL;
-	}
-
 	if (transfer && (transfer->speed_hz == 0))
 		req_hz = qspi->max_speed_hz;
 
@@ -514,10 +508,6 @@ static int zynq_qspi_setup_transfer(struct spi_device *qspi,
  */
 static int zynq_qspi_setup(struct spi_device *qspi)
 {
-
-	if (qspi->mode & SPI_LSB_FIRST)
-		return -EINVAL;
-
 	if (!qspi->max_speed_hz)
 		return -EINVAL;
 
-- 
1.7.5.4

